EIPS/eip-5920.md

```diff
@@ -0,0 +1,51 @@
+---
+eip: 5920
+title: PAY opcode
+description: Introduces a new opcode, PAY, to send ether to an address without calling it
+author: Pandapip1 (@Pandapip1)
+discussions-to: https://ethereum-magicians.org/t/eip-5920-pay-opcode/11717
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-03-14
+---
+
+## Abstract
+
+This EIP adds a new opcode, `PAY(addr, val)`, that transfers `val` wei to the address `addr` without calling it.
+
+## Motivation
+
+Currently, to send ether to an address requires you to call a function of that address, which has a few issues. First of all, it opens a reentrancy attack vector, as the recipient can call back into the sender. Secondly, it opens a DoS vector, so parent functions must be cognizant of the possibility that the recipient will run out of gas or revert. Finally, the `CALL` opcode is needlessly expensive for simple ether transfers, as it requires the memory and stack to be expanded, the recipient's full data including code and memory to be loaded, and finally needs to execute a call, which might do other unintentional operations. Having a dedicated opcode for ether transfers solves all of these issues, and would be a useful addition to the EVM.
+
+## Specification
+
+| Parameter | Value |
+| - | - |
+| `FORK_BLKNUM` | TBD |
+| `PAY_OPCODE` | TBD |
+| `HOT_GAS_COST` | `500` |
+| `COLD_GAS_COST` | `3000` |
+
+A new opcode is introduced: `PAY` (`PAY_OPCODE`), which:
+
+- Pops two values from the stack: `addr` then `val`.
+- Transfers `val` wei to the address `addr`.
+
+The cost of this opcode is `HOT_GAS_COST` if `addr` is a warm account, and `COLD_GAS_COST` if it is a cold account.
```

Comment:
[Peer review]

I think it's worthy of talking about gas costs' economy in the rationale when choosing the gas cost.

Technical context I could think of:

https://eips.ethereum.org/EIPS/eip-2929 which sets 

```
700 gas per BALANCE
```

```
COLD_ACCOUNT_ACCESS_COST 2600
```
as well as the yellow paper mention 

```
G_balance 400 Amount of gas to pay for a BALANCE operation
G_call 700 Paid for a CALL operation
```

Reply:
I was under the impression that the BALANCE opcode costs 100 gas. I could be wrong.

Reply:
I've updated the gas cost to reflect uninitialized accounts.

Definitely could use some motivation though. I'll keep this open.